CMakeLists.txt

```diff
@@ -85,6 +87,19 @@ add_compile_options(/WX /Gy
     # note that /Zi generates debug info inside the object file, it's the same as /Z7 for msvc
     "$<$<COMPILE_LANGUAGE:ASM_MASM>:/Zi;/W3;/nologo>")
 
+if(STL_USE_ANALYZE)
+    # warning C6553: The annotation for function 'LCMapStringEx' on _Param_(9)
+    # does not apply to a value type.
+    # There's a bug in the declaration for LCMapStringEx - it applies _In_opt_ to an LPARAM.
+    # LPARAM is a LONG_PTR (intptr_t), and it's invalid to apply _In_opt_ to a non-pointer.
+
+    # warning C28182: Dereferencing NULL pointer. 'Temp_value_#1457' contains the same NULL value as
+    # 'new(1*40, nothrow)' did.
+    # `new` expressions with allocation functions that may return `null` are incorrectly treated as dereferencing
+    # the new'd pointer. This is a bug in C28182.
+    add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:/analyze:autolog-;/wd6553;/wd28182>")

```

Comment:
Could we suppress these warnings more locally? I.e., do they fire in single TUs or everywhere?

Reply:
Half-done: I suppressed C28182 locally, as only 3 `unique_ptr` constructions in `tzdb.cpp` were affected, and that warning seems common/valuable.

C6553 is emitted by WinSDK headers, so we would have to wrap every inclusion. The warning seems less valuable, so I'm fine with global suppression.

Reply:
Thanks - that addresses my concerns.